src/dsymbol/scope_.d

```diff
@@ -215,6 +216,11 @@ struct Scope
 		return getSymbolsByName(name);
 	}
 
+	bool hasSymbolRecursive(const(DSymbol)* symbol) const
```

Comment:
this is not strictly part of the protection changes, I added it for the use-case of after checking protection in DCD